### PR TITLE
[macOS] Don't force the RID to be x64-only

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

Use maccatalyst-arm64 on arm64 and maccatalyst-x64 on x64.

### CODE REVIEW

**Review by Eilon — DISMISSED** [review]

**Comment by mattleibow** [comment]
Blazor is still causing issues with a parallel/double build of a RID for an app.

**Comment by mattleibow** [comment]
The real fix is here: https://github.com/dotnet/maui/pull/16765

**Review by StephaneDelcroix — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

File: src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <NoWarn>$(NoWarn),CA1416</NoWarn>
     <!-- Disable multi-RID builds to workaround a parallel build issue -->
     <RuntimeIdentifier Condition="$(TargetFramework.Contains('-maccatalyst'))">maccatalyst-x64</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="$(TargetFramework.Contains('-maccatalyst')) and '$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)' == 'arm64'">maccatalyst-arm64</RuntimeIdentifier>
```
Comment (Eilon) [inline src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj:12]: Does it matter that this is checking whether the _current machine_ is ARM64? Why would that matter for what RID to target? I get that it matters if you're going to _run_ it, but it shouldn't affect compilation, no?

Comment (mattleibow) [reply src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj:12]: Yeah, we typically build and run on the same machine on CI. The real issue is that Blazor and/or maccatalyst can do multiple rids. So without this people on arm64 can or deploy as it always builds x64.

**Review by Redth — APPROVED** [review]

**Pull request merged**